packages/replay/src/replay.ts

```diff
@@ -309,6 +367,37 @@ export class ReplayContainer implements ReplayContainerInterface {
     this.startRecording();
   }
 
+  /**
+   * If not in "session" recording mode, flush event buffer which will create a new replay.
+   * Unless `continueRecording` is false, the replay will continue to record and
+   * behave as a "session"-based replay.
+   *
+   * Otherwise, queue up a flush.
+   */
+  public async sendBufferedReplayOrFlush({ continueRecording = true }: SendBufferedReplayOptions = {}): Promise<void> {
+    if (this.recordingMode === 'session') {
+      return this.flushImmediate();
+    }
+
+    // Allow flush to complete before resuming as a session recording, otherwise
+    // the checkout from `startRecording` may be included in the payload.
+    // Prefer to keep the error replay as a separate (and smaller) segment
+    // than the session replay.
+    await this.flushImmediate();
+
+    if (!continueRecording) {
```

Comment:
Hmm wait a second, didn't a revision of this exist where we always called `this.stopRecording()` before, even if not continuing recording? I feel like we had a comment on that somewhere and you changed it actually 😅

Reply:
This branch isn't up-to-date with the base branch

Reply:
ahh oops 😅